cluster/gce/gci/configure-helper.sh

```diff
@@ -83,18 +148,124 @@ function safe-format-and-mount() {
   mount -o discard,defaults "${device}" "${mountpoint}"
 }
 
-# Local ssds, if present, are mounted at /mnt/disks/ssdN.
+# Creates a bind mounting from the /mnt/disks/ fs mounted folder to
+# a new mountpoint in /mnt/disks/by-uuid/ using the devices UUID 
+function unique-uuid-bind-mount(){
+  local device=$1
+  local interface=$2
+  local mountpoint=$3
+
+  # Trigger udev refresh so that newly formatted devices are propagated in by-uuid
+  udevadm control --reload-rules
+  udevadm trigger
+  udevadm settle
+
+  # Get the real UUID of the device
+  if [[ ${interface} == "nvme" ]]; then
+    # This condition is required because the existing Google images does not
+    # expose NVMe devices in /dev/disk/by-id so we are using the /dev/nvme instead
+    local item=$(echo ${device} | cut -d '/' -f 3)
+  else
+    local item=$(readlink -f ${device} | cut -d '/' -f 3)
+  fi
+  # item should be the kernel device name e.g. "sdb", "nvme0n1"
+  if [[ ${item} != sd* ]] && [[ ${item} != nvme* ]]; then
+    echo "'item' is not of the correct format. It must be the kernel name of the device, got ${item} instead" >&2
+    exit 1
+  fi
+
+  
+  local myuuid=$(ls -l /dev/disk/by-uuid/ | grep ${item} | tr -s ' ' | cut -d ' ' -f 9)
+  # myuuid should be the uuid of the device as found in /dev/disk/by-uuid/ 
+  if [[ -z ${myuuid} ]]; then
+    echo "Failed to get a uuid for device ${device} when mounting." >&2
+    exit 2
+  fi
+  
+  local bindpoint="${UUID_MNT_PREFIX}-${interface}-fs/local-ssd-${myuuid}"
+  # bindpoint should be the full path of the to-be-bound device
+
+  # Mount device to the mountpoint
+  mkdir -p "${bindpoint}"
+  echo "Binding '${device}' at '${bindpoint}'"
+  mount --bind "${mountpoint}" "${bindpoint}"
+  chmod a+w ${bindpoint}
+
+}
+
+# Mounts, bindmounts, or symlinks depending on the interface and format
+# of the incoming device
+function mount-ext(){
+  local ssd=$1
+  local interface=$2
+  local format=$3  
+
+  if [[ ${format} == "fs" ]]; then
+    if [[ ${interface} == "scsi" ]]; then
+      local scsinum=`echo ${ssd} | sed -e 's/\/dev\/disk\/by-id\/google-local-ssd-\([0-9]*\)/\1/'`
```

Comment:
We don't need to handle it now, but it may be good to think about how to handle partitioned disks.